templates/prometheus.go

```diff
@@ -73,69 +73,71 @@ var (
 
 var PrometheusTemplate = promv1.Prometheus{
 	Spec: promv1.PrometheusSpec{
-		ExternalLabels:         map[string]string{},
-		ServiceAccountName:     "prometheus-k8s",
-		ServiceMonitorSelector: &resourceSelector,
-		PodMonitorSelector:     &resourceSelector,
-		RuleSelector:           &resourceSelector,
-		EnableAdminAPI:         false,
-		Alerting: &promv1.AlertingSpec{
-			Alertmanagers: []promv1.AlertmanagerEndpoints{{
-				Namespace: "",
-				Name:      "alertmanager-operated",
-				Port:      intstr.FromString("web"),
-			}},
-		},
-		Resources:   utils.GetResourceRequirements("prometheus"),
-		ListenLocal: true,
-		Containers: []corev1.Container{{
-			Name: "kube-rbac-proxy",
-			Args: []string{
-				fmt.Sprintf("--secure-listen-address=0.0.0.0:%d", KubeRBACProxyPortNumber),
-				"--upstream=http://127.0.0.1:9090/",
-				"--logtostderr=true",
-				"--v=10",
-				"--tls-cert-file=/etc/tls-secret/tls.crt",
-				"--tls-private-key-file=/etc/tls-secret/tls.key",
-				"--client-ca-file=/var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt",
-				"--config-file=/etc/kube-rbac-config/config-file.json",
-			},
-			Ports: []corev1.ContainerPort{{
-				Name:          "https",
-				ContainerPort: int32(KubeRBACProxyPortNumber),
-			}},
-			VolumeMounts: []corev1.VolumeMount{
-				{
-					Name:      "serving-cert",
-					MountPath: "/etc/tls-secret",
+		CommonPrometheusFields: promv1.CommonPrometheusFields{
```

Comment:
- is the prom CRD installed in the cluster & prom-operator is compatible with this CR?